### DISCUSSION TITLE
Audio prop in Video component.

### DISCUSSION THREAD

**Discussion by xts-bit** [TheWidlarzGroup/react-native-video#4963] [Ideas]
### What problem are you trying to solve?

This feature can solve the problems of people who are trying to use react-native-video for displaying videos in a user-generated content app. It will completely eliminate the need to use FFmpeg for merging audio to video from the server -- which is very hard to do in a limited CPU and Ram. It will give bad memory errors while deploying.

### Proposed solution

A prop in the Video component for adding audio to a playing video.

**Possible implementation**

Decide on the API for this feature. You might introduce a new prop, such as `audioSource`, that allows users to specify an audio file or source URL. 2nd To ensure synchronization between audio and video, you may need to implement additional logic. One approach is to observe the `currentTime` of both the AVPlayer (for video) and the AVAudioPlayer (for audio) and adjust playback accordingly. Or simply get the video and audio duration as a prop.

### Additional context

**Code sample**

```
<Video source={{
     uri: "background.mp4",
     audioTracks="https://somerandomaudiourlfrominter.mp3"
  }}  
  ref={(ref) => {
    this.player = ref
  }}                                    
  onBuffer={this.onBuffer}               
  onError={this.videoError}           
  style={styles.backgroundVideo}
  
  />
```

**Comment by freeboub** [comment]
@xts-bit I updated the description and move the new prop *audioTracks* in source. I know, that on android this prop must be available at playback start up. I don't think it will make sense to change it during playback.
Notice that it would be great to move textTracks in source at the same time !

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
